### DISCUSSION TITLE
Sign commits made with codespaces via SSH signing key?

### DISCUSSION THREAD

**Discussion by thiloho** [community/community#55500] [Codespaces] [answered]
### Select Topic Area

Question

### Body

Is it possible to enable SSH key commit signing for GitHub Codespaces? The only thing I found was how to do it with GPG keys, as documented in [Managing GPG verification for GitHub Codespaces](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-gpg-verification-for-github-codespaces). As far as I can see, such setting is not available for the SSH signing.

I could create a new SSH key for each codespace and then add that to GitHub, but that is quite tedious and probably not the way you should do it.

Thank you in advance!

**Comment by samruddhikhandale** [comment, marked as answer]
Hi 👋 

I am curious, is there a specific scenario you are trying to cater by using SSH signing instead of [GPG verification](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-gpg-verification-for-github-codespaces)? 

Configuring GPG verification for Codespaces is simple as all you have to do is [enable this setting](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-gpg-verification-for-github-codespaces#enabling-or-disabling-gpg-verification).

> I could create a new SSH key for each codespace and then add that to GitHub, but that is quite tedious and probably not the way you should do it.

Alternatively, for SSH signing you could inject the private key as a user-level [codespaces secret](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-encrypted-secrets-for-your-codespaces). But a downside is that security is a concern, if you open a malicious repo in a codespace, it could steal your secrets. And if they steal your private key, they could sign any commit as you. So you'd only want to enable that user-level secret on repos that you trust.

As this secret would be set on repository level, you won't have to add it for every codespace. However, we would recommend switching to GPG verification (if possible)

**Reply by thiloho** [reply]
Hey

The scenario I am trying to "cater" is basically having to have two keys, one being the SSH key to seamlessly push without having to provide a password every time, and the other one being the GPG key.

I really liked the idea of having only one / the same key (SSH) to authenticate and sign the commits. Other than that, it is nothing, really.

**Reply by weakish** [reply]
> I really liked the idea of having only one / the same key (SSH) to authenticate and sign the commits.

@thiloho If you followed the instructions in GitHub documentation to configure GPG signing under Codespaces, commits in Codespaces will be signed using **GitHub's GPG key**, instead of *your* GPG key. So you still do not need to manage your GPG key on your local machine. On your locale machine, you can continue using only one key (SSH) to auth and sign.
